ui/src/kapacitor/components/TickscriptEditorConsole.js

```diff
@@ -1,22 +1,33 @@
 import React, {PropTypes} from 'react'
 
-const TickscriptEditorConsole = ({validation}) =>
-  <div className="tickscript-console">
-    <div className="tickscript-console--output">
-      {validation
-        ? <p>
-            {validation}
-          </p>
-        : <p className="tickscript-console--default">
-            Save your TICKscript to validate it
-          </p>}
+const TickscriptEditorConsole = ({consoleMessage, unsavedChanges}) => {
+  let consoleOutput
+  let consoleClass = 'tickscript-console--default'
+
+  if (!unsavedChanges) {
+    consoleOutput = 'TICKscript is valid'
+    consoleClass = 'tickscript-console--valid'
+  } else if (unsavedChanges && !consoleMessage) {
```

Comment:
At this point `unsavedChanges` is guaranteed. This chain should be

```
if (!unsavedChanges) {

} else if (!consoleMessage) {

} else {

}
```

And actually preferably, rephrase in the positive:

```
if (consoleMessage) {
  // tickscript-console--error
} else if (unsavedChanges) {
  // tickscript-console--default
} else {
  // tickscript-console--valid
}
```

Reply:
Good catch! Needed fresh eyes on this